src/sync.rs

```diff
@@ -94,8 +99,37 @@ impl<A> SyncArbiter<A>
 where
     A: Actor<Context = SyncContext<A>>,
 {
-    /// Start new sync arbiter with specified number of worker threads.
-    /// Returns address of the started actor.
+    /// Start new `SyncArbiter` with specified number of worker threads.
+    /// Returns the address of the started actor pool.
+    ///
+    /// ## Example
```

Comment:
When people come into the docs via docs.rs, or are pasting a link, they'll see something like this today:

https://docs.rs/actix/0.7.10/actix/sync/struct.SyncArbiter.html

There *are* no examples, and no guides on how to use the function. When you get a link to start directly:

https://docs.rs/actix/0.7.10/actix/sync/struct.SyncArbiter.html#method.start

Then it's important to have the example right there showing the usage.

I'd rather keep all the examples in here please, especially given that documentation and accessibility seem to be big barriers to entry to the project at the moment.

Reply:
Then please make link to sync module instead. You do realize that each example is going through doc test(which slows down `cargo test`).

There is no need to repeat the same example if you could make hyper link like
```
Please refer to [module](../index.html) documentation for example
```

Reply:
I don't think that this is a justification to remove examples? Saying that they go through tests that causes a slow down seems like we are optimising for a test time, rather than saying "let's give the best documentation possible". It also means people have to play link-diving, and that doesn't aid accessibility of relevant information for users.

I think that examples also have to be focused, and specific to the details we are trying to highlight. Continually pointing at one "mega example" will add confusion, not clarity to the documentation we are trying to show, because people won't know anymore what is or isn't relevant to the goal they are trying to achieve.

I also want to say, these functions may be obvious to you as a library expert, but they are not obvious to others. There are often many questions on gitter and issues raised about confusion on how to operate the library correctly.

Reply:
Adding reference on how to use sync actor, should prompt user to click and go read the full example instead of this short one.

I don't understand how asking to click link should make it confusing?

Reply:
Links always confuse people, because you remove context of the documentation from the example which contains a great variety of components. Then it becomes a question of "what is relevant here?".

I think in this case, because it's construction, okay, this example can go. But if there are other cases, I want there to be examples embedded to the docs. For example, the rust docs have examples like https://doc.rust-lang.org/std/result/enum.Result.html#method.or_else on all their methods, rather than referring to a singular example. 

Does that seem like an acceptable approach?

Reply:
If method is complex enough, for example it has generic bounds, example would be helpful to clear our it usage.
But for simple methods with clear semantics, I do not see a need for example

Reply:
I think we'll have to "discuss" when we come to these "simple vs complex" methods, because I still think there is a gap in our expectations of simple vs complex (IE I can't work out what a method does from type signatures, I need examples generally, where as others can read type signatures and know exactly what's happening.

So in this case I'll remove this specific example.